metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -290,20 +291,28 @@ def visit_join_over_time_range_node(self, node: JoinOverTimeRangeNode) -> SqlDat
         input_data_set = node.parent_node.accept(self)
         input_data_set_alias = self._next_unique_table_alias()
 
-        agg_time_dimension_instance: Optional[TimeDimensionInstance] = None
+        # Find requested agg_time_dimensions in parent instance set.
+        # For now, will use instance with smallest granularity in time spine join.
+        # TODO: use metric's default_grain once that property is available.
```

Comment:
Is that what we want? I think we need to always join on the smallest selected. If default grain is DAILY and someone selects `metric_time__hourly, metric_time__daily` I think we still need to join on hourly. But of course that's only true if hourly is requested. Or maybe it should be whatever the window grain is set to? I'm confused.

Anyway, that's a tomorrow problem, doesn't affect this PR.

Reply:
I think what it will end up being is: use the default unless that's too big based on the selected grains, in which case use the largest grain allowed for the selected grains.